src/sorting/src/bubble_sort.cairo

```diff
@@ -5,39 +5,44 @@
 /// * `array` - Array to sort
 /// # Returns
 /// * `Array<usize>` - Sorted array
-pub fn bubble_sort_elements<T, +Copy<T>, +Drop<T>, +PartialOrd<T>, +PartialEq<T>>(
-    mut array: Array<T>, asc: bool
-) -> Array<T> {
-    if array.len() <= 1 {
-        return array;
-    }
-    let mut idx1 = 0;
-    let mut idx2 = 1;
-    let mut sorted_iteration = 0;
-    let mut sorted_array = array![];
+use super::Sortable;
+
+pub impl BubbleSort of Sortable {
+    fn sort<T, +Copy<T>, +Drop<T>, +PartialOrd<T>>(mut array: Span<T>) -> Array<T> {
+        if array.len() == 0 {
+            return array![];
+        }
+        if array.len() == 1 {
+            return array![*array[0]];
+        }
+        let mut idx1 = 0;
+        let mut idx2 = 1;
+        let mut sorted_iteration = true;
+        let mut sorted_array = array![];
 
-    loop {
-        if idx2 == array.len() {
-            sorted_array.append(*array[idx1]);
-            if sorted_iteration == 0 {
-                break;
-            }
-            array = sorted_array;
-            sorted_array = array![];
-            idx1 = 0;
-            idx2 = 1;
-            sorted_iteration = 0;
-        } else {
-            if (*array[idx1] == *array[idx2]) || !((asc) ^ (*array[idx1] < *array[idx2])) {
+        loop {
+            if idx2 == array.len() {
                 sorted_array.append(*array[idx1]);
-                idx1 = idx2;
-                idx2 += 1;
+                if sorted_iteration {
```

Comment:
isn't this a while ? (do we want to keep the bubble sort ?)

Reply:
wdym?
Putting this as the while condition to exit?

Reply:
yes

Reply:
can't because it also depends on whether `idx2 == array.len()`.
Or maybe we can?
Didn't try much to update the logic in this case